app/Libraries/Payments/PaypalCreatePayment.php

```diff
@@ -110,6 +126,20 @@ private function getRedirectUrls()
             ->setCancelUrl(route('payments.paypal.declined', ['order_id' => $this->order->order_id]));
     }
 
+    private function getShippingAddress()
+    {
+        $address = $this->order->address;
+
+        return (new ShippingAddress())
```

Comment:
A quick check shows that passing array works as well instead of calling all the functions (also applies to all other paypal objects?).

Reply:
and we're not going to be using any of it

Reply:
?_?

I mean something like `return new ShippingAddress(['city' => ...])`.